src/Encoder/EncoderPropertiesTrait.php

```diff
@@ -201,6 +201,8 @@ protected function getUrlPrefix(): string
      */
     public function withIncludedPaths(iterable $paths): EncoderInterface
     {
+        $paths = $this->iterableToArray($paths);
+
         \assert(
             \call_user_func(
                 function (array $paths): bool {
```

Comment:
Unfortunately, this cannot reliable be changed to `iterable` and work with `Generator` instances, because generators can only be iterated once and the `assert` (at least in development) will be run before the included paths can be used.

The best option is to immediately convert to an array, as per the change above.